File: app/src/main/java/com/firebaseapp/sowbreira_26fe1/fl_mane/ImageAdapter.java
package com.firebaseapp.sowbreira_26fe1.fl_mane;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;

import com.squareup.picasso.Picasso;

import jp.wasabeef.picasso.transformations.CropCircleTransformation;

public class ImageAdapter extends BaseAdapter {
    private Context mContext;

    public ImageAdapter(Context c) {
        mContext = c;
    }

    public int getCount() {
        return 12;
    }

    public Object getItem(int position) {
        return null;
    }

    public long getItemId(int position) {
        return 0;
    }

    // create a new ImageView for each item referenced by the Adapter
    public View getView(int position, View convertView, ViewGroup parent) {
        ImageView imageView;
        if (convertView == null) {
            // if it's not recycled, initialize some attributes
            imageView = new ImageView(mContext);
            imageView.setLayoutParams(new ViewGroup.LayoutParams(200, 200));
            imageView.setScaleType(ImageView.ScaleType.CENTER_CROP);
            imageView.setPadding(8, 8, 8, 8);
        } else {
            imageView = (ImageView) convertView;
        }

        String path = "https://sowbreira-26fe1.firebaseapp.com/f1mane/profile/profile-"+position+".png";
        Picasso.with(mContext).load(path)
                .transform(new CropCircleTransformation())
                .placeholder(R.drawable.ic_user_place_holder)
                .into(imageView);
        return imageView;
    }


}

File: app/src/main/java/com/firebaseapp/sowbreira_26fe1/fl_mane/ImageAdapter.kt
package com.firebaseapp.sowbreira_26fe1.fl_mane

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView

import coil3.load
import coil3.request.placeholder
import coil3.request.transformations
import coil3.transform.CircleCropTransformation

class ImageAdapter(private val mContext: Context) : BaseAdapter() {

    override fun getCount(): Int = 12

    override fun getItem(position: Int): Any? = null

    override fun getItemId(position: Int): Long = 0

    // create a new ImageView for each item referenced by the Adapter
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val imageView: ImageView
        if (convertView == null) {
            // if it's not recycled, initialize some attributes
            imageView = ImageView(mContext)
            imageView.layoutParams = ViewGroup.LayoutParams(200, 200)
            imageView.scaleType = ImageView.ScaleType.CENTER_CROP
            imageView.setPadding(8, 8, 8, 8)
        } else {
            imageView = convertView as ImageView
        }

        val path = "https://sowbreira-26fe1.firebaseapp.com/f1mane/profile/profile-$position.png"
        imageView.load(path) {
            placeholder(R.drawable.ic_user_place_holder)
            transformations(CircleCropTransformation())
        }
        return imageView
    }
}
